pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -150,4 +162,46 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
     }
     return pinotTaskConfigs;
   }
-}
+
+  /**
+   * Fetch completed (DONE/UPLOADED) segment and partition information
+   *
+   * @param realtimeTableName the realtime table name
+   * @param completedSegmentsZKMetadata list for collecting the completed (DONE/UPLOADED) segments ZK metadata
+   * @param partitionToLatestLLCSegmentName map for collecting the partitionId to the latest LLC segment name
+   * @param allPartitions set for collecting all partition ids
+   */
+  private void getCompletedSegmentsInfo(String realtimeTableName, List<SegmentZKMetadata> completedSegmentsZKMetadata,
+      Map<Integer, String> partitionToLatestLLCSegmentName, Set<Integer> allPartitions) {
+    List<SegmentZKMetadata> segmentsZKMetadata = _clusterInfoAccessor.getSegmentsZKMetadata(realtimeTableName);
+
+    Map<Integer, LLCSegmentName> latestLLCSegmentNameMap = new HashMap<>();
+    for (SegmentZKMetadata segmentZKMetadata : segmentsZKMetadata) {
+      CommonConstants.Segment.Realtime.Status status = segmentZKMetadata.getStatus();
+      if (status.isCompleted()) {
+        completedSegmentsZKMetadata.add(segmentZKMetadata);
+      }
+
+      // Skip UPLOADED segments that don't conform to the LLC segment name
```

Comment:
I don't think we want to skip the uploaded segments even if the name doesn't conform to the LLC segment name

Reply:
+1 We want purge jobs to handle real time segments uploaded. cc @sajjad-moradi

Reply:
Yes remove this never used part !